sklearn/utils/tests/test_utils.py

```diff
@@ -472,9 +473,9 @@ def test_safe_indexing_pandas_no_settingwithcopy_warning():
 
     X = pd.DataFrame({"a": [1, 2, 3], "b": [3, 4, 5]})
     subset = _safe_indexing(X, [0, 1], axis=0)
-    with pytest.warns(None) as record:
+    with warnings.catch_warnings():
+        warnings.simplefilter("error", SettingWithCopyWarning)
```

Comment:
`pd` is already imported here using `pytest.importorskip("pandas")` a few lines above.

```suggestion
        warnings.simplefilter("error", pd.core.common.SettingWithCopyWarning)
```